text/0000-api-version-compat.md

```diff
@@ -0,0 +1,122 @@
+As the community of platform and buildpack authors grows and matures, this pain is magnified.
+
+Now that the APIs have stabilized we can realistically guarantee that we will not make breaking lifecycle changes before the implementation of the 1.0 APIs are released.
+
+## Deprecations
+We will allow the spec to remove features in a minor API and interpret this as a deprecated feature in the lifecycle.
+Previously, we never formalized rules around deprecations. This made sense because all API changes have been breaking.
+However, deprecations can aid platforms and buildpacks as they prepare to perform a major API upgrade.
+
+The lifecycle will support deprecation modes to allow platform/buildpack authors to curate the experience for users
+and/or filter for deprecations relevant to their use case.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Spec
+The spec will use future `0.x` API release to indicate changes that can be implemented in a non-breaking fashion. These may include:
+- purely additive features
+- simple flag/file/key renames (the previous usage will be supported but deprecated by the lifecycle)
+- removal of features (these will be interpreted as deprecated by the lifecycle)
+
+## Lifecycle
+The lifecycle will continue to use the [lifecycle descriptor](https://github.com/buildpacks/rfcs/blob/pack-publish-buildpack/text/0011-lifecycle-descriptor.md)
+to indicate the implemented API versions as before. But the assumptions we make about compatibility will change.
+
+## Deprecations
+If a feature is removed from the lifecycle's advertised API version but used by a platform or buildpack the lifecycle will print a deprecation warning.
+Because platform authors may want to shield end users from deprecation warning caused by the platform's use of deprecated
+lifecycle features, the lifecycle will provide the following options to configure deprecation warnings:
+
+All lifecycle binaries will have a deprecation mode, configurable in the following way:
+| Flag              | Environment Variable | Default
+|-------------------|----------------------|--------
+| -deprecation-mode | CNB_DEPRECATION_MODE | warn
+
+The allowed values of deprecation mode are as follows:
+| Mode                | Behavior
+|---------------------|---------
+| all                 | prints all deprecation warnings
+| buildpack           | prints only buildpack API deprecation warnings
+| platform            | prints only platform API deprecation warnings
+| none                | silences all deprecation warnings
+| fail                | prints an error message and fail when any deprecated feature is used
+| fail-buildpack      | fail for buildpack deprecations
+| fail-platform       | fail for platform deprecations
+
+# How it Works
+[how-it-works]: #how-it-works
+## Spec
+The core team will ensure that no changes are added to a minor spec release that imply incompatibility with the previous minor.
```

Comment:
The expectation being that deprecated features can be explicitly removed at `1.0`?

Reply:
I know this has changed a bit since you left this comment, but, for posterity: Deprecated features will be removed from the 1.0 API. Platforms & buildpacks will not be able to use these features when using the 1.0 API (the lifecycle itself may continue to support them in older API "modes", even after the 1.0 API is supported).